internal/optimization_test.go

```diff
@@ -0,0 +1,90 @@
+package internal
```

Comment:
Can you please move this test to `download_test.go`. This is the place where we test all high-level download use-cases.